src/query/plan/operator.cpp

```diff
@@ -4527,7 +4537,8 @@ class CallProcedureCursor : public Cursor {
         // result_ needs to live throughout multiple Pull evaluations, until all
         // rows are produced. Therefore, we use the memory dedicated for the
         // whole execution.
-        result_(nullptr, mem) {
+        result_(utils::Allocator<mgp_result>(self_->memory_resource)
```

Comment:
Is the comment above still correct? Should we use the mem argument?

Reply:
I think it is still okay. Using different memory is key here for doing deallocations by just clearing that specific memory for CallProcedure. As long as CallProcedureCursors lives, so will CallProcedure live and so will this dedicated memory. I will just update comment.